app/src/main/java/com/bogdan/codeforceswatcher/features/problems/ProblemsFragment.kt

```diff
@@ -156,12 +133,14 @@ private fun ContentView(
     topBar = { NavigationBar(onFilter, onSearch) },
     backgroundColor = AlgoismeTheme.colors.primaryVariant
 ) {
+    if (problemsState.value == null) return@Scaffold
+
     SwipeRefresh(
-        state = rememberSwipeRefreshState(isRefreshState.isRefreshing),
+        state = rememberSwipeRefreshState(problemsState.value!!.status != ProblemsState.Status.IDLE),
         onRefresh = { onRefresh() }
     ) {
         ProblemsList(
-            problemsState = problemsState,
+            problemsState = problemsState as State<ProblemsState>,
```

Comment:
Can't we pass just list here?

Reply:
ProblemsState in ProblemsList is not optional, so it needed a cast.

Reply:
I just fixed it too, applied your comments above.